Source/JavaScriptCore/wasm/WasmBBQIRGenerator.cpp

```diff
@@ -0,0 +1,3383 @@
+
+            // Ensure all non-constant parameters are loaded into registers.
+            if (!lhs.isConst())
+                lhsLocation = loadIfNecessary(currentControlData(), lhs);
+            if (!rhs.isConst())
+                rhsLocation = loadIfNecessary(currentControlData(), rhs);
+
+            ASSERT(lhs.isConst() || lhsLocation.isRegister());
+            ASSERT(rhs.isConst() || rhsLocation.isRegister());
+            
+            kill(lhs); // If either of our operands are temps, kill them and liberate any bound
+            kill(rhs); // registers. This lets us reuse one of the registers for the output.
+
+            Location toReuse = lhs.isConst() ? rhsLocation : lhsLocation; // Select the location to reuse, preferring lhs.
+
+            result = Value::fromTemp(resultType, m_parser->expressionStack().size()); // Result will be the new top of the stack.
+            Location resultLocation = tryBind(result, toReuse);
+            ASSERT(resultLocation.isRegister());
+
+            if (lhs.isConst() || rhs.isConst())
+                regImm(lhs, lhsLocation, rhs, rhsLocation, resultLocation);
+            else 
+                regReg(lhs, lhsLocation, rhs, rhsLocation, resultLocation);
+
+            return { };
+        }
+    }
+
+    template<typename Fold, typename Reg>
+    inline PartialResult unary(TypeKind resultType, Value& operand, Value& result, Fold fold, Reg reg) {
+        if (operand.isConst()) {
+            result = fold(operand);
+            return { };
+        } else {
+            Location operandLocation = loadIfNecessary(currentControlData(), operand);
+            ASSERT(operandLocation.isRegister());
+
+            kill(operand); // If our operand is a temp, kill it and liberate its register if it has one.
+
+            result = Value::fromTemp(resultType, m_parser->expressionStack().size()); // Result will be the new top of the stack.
+            Location resultLocation = tryBind(result, operandLocation); // Try to reuse the operand location.
+            ASSERT(resultLocation.isRegister());
+
+            reg(operand, operandLocation, resultLocation);
+            return { };
+        }
+    }
+
+    struct ImmHelpers {
+        ALWAYS_INLINE static Value& imm(Value& lhs, Value& rhs)
+        {
+            return lhs.isConst() ? lhs : rhs;
+        }
+
+        ALWAYS_INLINE static Location& immLocation(Location& lhsLocation, Location& rhsLocation)
+        {
+            return lhsLocation.isRegister() ? rhsLocation : lhsLocation;
+        }
+
+        ALWAYS_INLINE static Value& reg(Value& lhs, Value& rhs)
+        {
+            return lhs.isConst() ? rhs : lhs;
+        }
+
+        ALWAYS_INLINE static Location& regLocation(Location& lhsLocation, Location& rhsLocation)
+        {
+            return lhsLocation.isRegister() ? lhsLocation : rhsLocation;
+        }
+    };
+
+    #define BLOCK(...) __VA_ARGS__
```

Comment:
Why?

Reply:
It's meant to shield any commas in the parameters of `EMIT_UNARY` and `EMIT_BINARY` from being interpreted as macro parameter separators. It's probably unnecessary for most opcode implementations, since most of the time commas only occur within (function) parameter lists anyway, but it prevents weirdness with multiple template parameters (i.e. `emitModOrDiv<int32_t, false>(...)`) or comma expressions if they do occur. 

It is worth evaluating if there's any case where this is truly necessary though, as opposed to something simpler like a curly-braced block. I'll give it a look! 👀